app/models/namespace_bot.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+# entity class for NamespaceBot
+class NamespaceBot < ApplicationRecord
+  has_logidze
+  acts_as_paranoid
+
+  belongs_to :user, class_name: 'User', dependent: :destroy
+  belongs_to :namespace, class_name: 'Namespace'
+
+  validates :user_id, uniqueness: { scope: [:namespace_id] }
+
+  validate :validate_bot_user_type
+
+  def validate_bot_user_type
+    return unless User.user_types[user.user_type] == User.user_types[:human]
```

Comment:
**nitpick (blocking):** I think you can just do `return unless user.human?` since we are using enums